python/private/pypi/pypi_repo_utils.bzl

```diff
@@ -51,7 +51,20 @@ def _resolve_python_interpreter(mrctx, *, python_interpreter = None, python_inte
     python_interpreter = _get_python_interpreter_attr(mrctx, python_interpreter = python_interpreter)
 
     if python_interpreter_target != None:
-        python_interpreter = mrctx.path(python_interpreter_target)
+        # The following line would make the MODULE.bazel.lock platform
+        # independent, because the lock file will then contain a hash of the
+        # file so that the lock file can be recalculated, hence the best way is
+        # to add this directory to PATH.
+        #
+        # hence we add the root BUILD.bazel file and get the directory of that
+        # and construct the path differently. At the end of the day we don't
+        # want the hash of the interpreter to end up in the lock file.
+        if hasattr(python_interpreter_target, "same_package_label"):
```

Comment:
FYI @UebelAndre, this may help in https://github.com/bazelbuild/rules_rust/issues/3297